### ISSUE TITLE
Batch Size =1 

### ISSUE THREAD

**Issue by Mohit-robo** [videturfortuna/vehicle_reid_itsc2023#10]
The code throws an error with batch_size = 1
`IndexError: too many indices for tensor of dimension 1`

**Comment by videturfortuna** [comment]
Yeah, for training you require batches with PK sampling so 1 is not a valid option, for inference perhaps u need to change some line of code...

**Comment by 22ema** [comment]
I fix the code. @videturfortuna. Could you check pull requests?
@Mohit-robo Could you check pull requests code? it's fix that error.

https://github.com/videturfortuna/vehicle_reid_itsc2023/pull/9#issue-2172754255

**Comment by Mohit-robo** [comment]
> I fix the code. @videturfortuna. Could you check pull requests? @Mohit-robo Could you check pull requests code? it's fix that error.
> 
> [#9 (comment)](https://github.com/videturfortuna/vehicle_reid_itsc2023/pull/9#issue-2172754255)

Sure thanks, let me check and I'll let you know. But did you check whether the results changed or no?

**Comment by 22ema** [comment]
@Mohit-robo 
Yes, when I debugged, I confirmed that the emb had a shape of (1,2048), not (2048).

And the code has been merged. We recommend doing a git pull.

**Comment by Mohit-robo** [comment]
> @Mohit-robo Yes, when I debugged, I confirmed that the emb had a shape of (1,2048), not (2048).
> 
> And the code has been merged. We recommend doing a git pull.

Sure thanks

**Comment by Mohit-robo** [comment]
Hey, thanks for the solution. Also is there any way I could use your pre-trained weights and fine-tune them on my datasets.
